src/triage/component/catwalk/storage.py

```diff
@@ -213,123 +243,179 @@ def matrix_with_sorted_columns(self, columns):
                     Columnset and desired columnset mismatch. Unique items: %s
                 ''', columnset ^ desired_columnset)
 
-    def save_yaml(self, df, project_path, name):
-        with smart_open.smart_open(os.path.join(project_path, name + ".yaml"), "wb") as f:
-            yaml.dump(df, f, encoding='utf-8')
+    def save_metadata(self, df, project_path, name):
+        path_parsed = urlparse(project_path)
+        scheme = path_parsed.scheme  # If '' of 'file' is a regular file or 's3'
+        if not scheme or scheme == 'file':  # Local file
+            with open(os.path.join(project_path, name + ".yaml"), "wb") as f:
+                yaml.dump(df, f, encoding='utf-8')
+        elif scheme == 's3':
+            s3 = s3fs.S3FileSystem()
+            with s3.open(os.path.join(project_path, name + ".yaml"), "wb") as f:
+                yaml.dump(df, f, encoding='utf-8')
+        else:
+            raise ValueError(f"""
+                  URL scheme not supported:
+                  {scheme} (from {os.path.join(project_path, name + '.yaml')})
+            """)
```

Comment:
Certainly not a big deal, but that's going to be quite the error message to read out of the logs. if it's important to organize it in the code this way, you might use `textwrap.dedent` –

    raise ValueError(textwrap.dedent(f"""\
        …
        …"""
    )

But I think you can cleanly write –

    raise ValueError("URL scheme not supported: "
                     f"{scheme} (from {path})")

(And, rather than repeat your path construction throughout the method, do it once at the top.)